### PR TITLE
Actual state machine shooter, fix to autonomous

### PR DESCRIPTION
**Pull request by cannedlock**

### CODE REVIEW

File: components/ShooterLogic.py [hunk 1]
```diff
@@ -34,127 +34,105 @@ def on_enable(self):
...
-        self.next_state('autoLoading')
-
     @state
-    def autoLoading(self, state_tm):
-        """Logic for running loader automatically"""
-        loaderSensor = 4
-        if not self.SensorArray[loaderSensor].get() and self.SensorArray[0].get():
-            self.shooterMotors.runLoader(self.loaderMotorVal)
-            self.sleepTime = .25
+    def checkForBall(self):
```
Comment (tuffinmuffin) [inline components/ShooterLogic.py]: checkForBall in which sensor? Should anything be running?

File: components/ShooterLogic.py [hunk 2]
```diff
@@ -34,127 +34,105 @@ def on_enable(self):
...
+    def loadBall(self):
+        """Loads ball if ball has entered."""
+        self.shooterMotors.runLoader(self.loaderMotorVal)
+        self.next_state('stopBall')
 
-        if self.SensorArray[loaderSensor].get():
-            sleep(self.sleepTime)
+    @state
+    def stopBall(self):
+        """Stops the loader if conditions are met."""
```
Comment (tuffinmuffin) [inline components/ShooterLogic.py]: stopBall? Not descriptive and comment does not explain it.

File: components/ShooterLogic.py [hunk 3]
```diff
@@ -34,127 +34,105 @@ def on_enable(self):
...
     @state
-    def setupShootAutomatically(self):
-        """Predecessor to automatic shooting"""
+    def initShooting(self):
+        """Smart shooter initialization (reversing if necessary)."""
         if not self.SensorArray[0].get():
-            self.shooterMotors.runLoader(-.3)
+            self.shooterMotors.runLoader(-self.loaderMotorVal)
 
         elif self.SensorArray[0].get():
```
Comment (tuffinmuffin) [inline components/ShooterLogic.py]: We should make the top and bottom sensors index a variable so it is descripve which one you are looking at. int enum may work well for this.

File: components/ShooterLogic.py [hunk 4]
```diff
@@ -34,127 +34,105 @@ def on_enable(self):
...
         self.shooterMotors.runShooter(1)
-        self.logger.debug("running auto shooter")
-        if state_tm > 1:
-            self.shooterMotors.runLoader(self.loaderMotorVal)
-            if state_tm > 4:
-                self.shooterMotors.stopLoader()
-                self.shooterMotors.stopShooter()
-                self.currentSensor = 3
-                self.done()
+        if self.shooterMotors.shooterMotor.getEncoder().getVelocity() >= 4800 or state_tm > 2:
```
Comment (tuffinmuffin) [inline components/ShooterLogic.py]: Make the 2 second be a variable with meaninfulg name. Same for target RPM.

File: components/ShooterLogic.py (hunk 1 above)
Comment (cannedlock) [reply components/ShooterLogic.py]: added more to docstring

File: components/ShooterLogic.py (hunk 2 above)
Comment (cannedlock) [reply components/ShooterLogic.py]: added more to docstrings

File: components/ShooterLogic.py (hunk 3 above)
Comment (cannedlock) [reply components/ShooterLogic.py]: added enum class to enumerate both sensors used

**Pull request merged**